pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -661,9 +661,15 @@ def _handle_image_build_failures(self):
     async def _build_images(self):
         self.runtime.logger.info('Building images')
 
+        group_config = await util.load_group_config(
+            group=f'openshift-{self.version.stream}',
+            assembly=self.assembly,
+            doozer_data_path=self.data_path
+        )
         signing_mode = await util.get_signing_mode(
```

Comment:
I think the best way to fix this is provide `get_signing_mode` function with `assembly` and `data_path` params, and let it load the group config if needed

Reply:
the get_signing_mode function will use passed group_config as a key parameter, so if group_config not passed, it will use group and assembly to load again, let's make the explicit loading outside the function

Reply:
This is exactly what I mean: `get_signing_mode` should be updated to accept assembly and data path, so it can work effectively against forks/assemblies. It could be useful in other places, and we wouldn't need to load group config outside it every time

Reply:
the get_signing_mode is just got software_lifecycle from config then do a translate, maybe we can define a config in group.yml like
```
software_lifecycle:
  phase: pre-release
  signing: unsigned
```
and then get rid of the function

Reply:
In this case, I think `software_lifecycle` might just go away. AFAIK, we're only using the pre-release/release/eol fields to decide about signing mode.

Reply:
Although, I would rather do this in a separate task, because we'd need to change lots of places in the code, and group config for all versions.

Reply:
ok I will create a separate task for signing mode [ART-8480](https://issues.redhat.com/browse/ART-8480)